packages/rs-drive-abci/src/execution/initialization.rs

```diff
@@ -82,9 +90,28 @@ where
         Ok(ResponseInitChain {
             consensus_params: None, //todo
             app_hash: app_hash.to_vec(),
-            validator_set_update: None,
+            validator_set_update: Some(validator_set),
             next_core_chain_lock_update: None,
-            initial_core_height: fork_info.since, // we send back the core height when the fork happens
+            initial_core_height: core_height, // we send back the core height when the fork happens
         })
     }
+
+    /// Determine initial core height.
+    ///
+    /// TODO: rewrite this, it is non-deterministic
+    /// We use either core height received from Tenderdash (from genesis file), OR the current tip of active core chain.
+    /// We use current tip as default because we need a fully functional, up-to-date validator set.
+    fn initial_core_height(
+        &self,
+        requested: u32,
+        fork_info: &Bip9SoftforkInfo,
+    ) -> Result<u32, Error> {
+        let core_height = if requested != 0 {
+            requested
+        } else {
+            fork_info.since
+        };
+
+        Ok(core_height)
+    }
```

Comment:
Not sure I understand the point here. Let's imagine we send core a height 0 -> it won't work. Let's imagine we send a height it doesn't know about -> non deterministic. I think we would never want the requested height.